Mathlib/Data/Nat/Factorial/SuperFactorial.lean

```diff
@@ -0,0 +1,49 @@
+/-
+Copyright (c) 2023 Moritz Firsching. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Moritz Firsching
+-/
+import Mathlib.Data.Nat.Factorial.Basic
+
+/-!
+# Super factorial
+
+This file defines the super factorial `1! * 2! * 3! * ...* n!`.
+
+## Main declarations
+
+* `Nat.superFactorial`: The super factorial.
+-/
+
+
+namespace Nat
+
+/-- `Nat.superFactorial n` is the super factorial of `n`. -/
+@[simp]
+def superFactorial : ℕ → ℕ
+  | 0 => 1
+  | succ n => factorial n.succ * superFactorial n
+
+/-- `sf` notation for super factorial -/
+scoped notation  "sf" n => Nat.superFactorial n
```

Comment:
https://mathworld.wolfram.com/Superfactorial.html suggests `$` notation; I think a source for this notation would be good, even if the answer is just wikipedia.

Reply:
I don't particularly care which notation we use, will add a reference in any case. Perhaps I make a poll on zulip?

Reply:
Is it even possible to use `$` as a notation? I thought it's reserved as a alias of `<|`.

Reply:
Even if it was possible, it would be at least confusing. So let's stick with `sf` I guess...